src/util/StoreProperty.js

```diff
@@ -0,0 +1,53 @@
+import { isObjectLike } from './AttributeParser';
+
+export class StoreProperty {
+	_observer = new Set();
+	_singlePropertyMode = false;
+	_state = {};
+
+	constructor(value) {
+		// wrap primitives with a generic "value" field to generate getter and setter
+		const specificValues = isObjectLike(value) ? value : { value };
+		this._state = { ...this.properties(), ...specificValues };
+		this._singlePropertyMode = Object.keys(this._state).length === 1;
```

Comment:
@quarkus Nice! But I don't like that this will happen automagically behind the scenes... What if I want my Store to only have one property - but it should be a nested property and not hoisted?

Reply:
Maybe we could check if `value` given via the constructor `isObjectLike` and if not activate the single property mode?